### PR TITLE
BuildInfo POJOs Spring annotations

### PR DESCRIPTION
**Pull request by MagoDelBlocco**
Added all needed Spring Cloud Datastore annotations for smooth GCD integration to BuildInfo and all related subclasses, and changed the timestamp field in said classes to google.cloud.Timestamp, for uniformity among all storage components.

Annotations added:
* @ Entity(name = \<kind\>) to BuildInfo and all subclasses.
* \@ Field and @ Id to BuildInfo fields.
* \@ Unindexed to builders field, since we don't need to query the database on that.
For more info:
https://cloud.spring.io/spring-cloud-static/Greenwich.RC1/multi/multi__spring_data_cloud_datastore.html

google.cloud.Timestamp:
Is the data format that can be read and transcribed into what Google Cloud Datastore uses internally. For more info: https://googleapis.dev/java/google-cloud-clients/0.92.0-alpha/com/google/cloud/datastore/TimestampValue.html

note: these files have already been reviewed in #36 . Only the Spring-specific annotations and the timestamp data type have been added on top of that.

### CODE REVIEW

File: src/main/java/com/google/graphgeckos/dashboard/storage/Builder.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+
+/**
+ * Represents the information about a single buildbot.
+ */
+@Entity(name = "builder")
+public class Builder {
+  // name of the builder
+  private final String builderName;
+  // builder logs
```
Comment (samuelsogbesan) [inline src/main/java/com/google/graphgeckos/dashboard/storage/Builder.java:14]: Please capitalise the start of comments. The [style guide](https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment) refers to Jdoc here but this should apply to regular comments too

File: src/main/java/com/google/graphgeckos/dashboard/storage/Builder.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  private final List<Log> logs;
+
+  // builder status, FAILED or PASSED
+  private final BuilderStatus status;
```
[the remark this answers is not in the document]
Comment (MagoDelBlocco) [reply src/main/java/com/google/graphgeckos/dashboard/storage/Builder.java:18]: Thank you for pointing this out! I believe we should change this to only "builder status, as described in the BuilderStatus class". We might expand the states BuilderStatus might have, so this makes sense in the long run.

**Comment by MagoDelBlocco** [comment]
This PR is getting too bloated with changes from 4 different classes. Will be split into 3 different PRs as follows:
This one: changes to BuildInfo
Number 2: changes to Builder
Number 3: Log & BuilderStatus changes

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  @Field(name = "commitHash")
+  private final String commitHash;
+
+  @Field(name = "timestamp")
+  private final Timestamp timestamp;
+
+  @Field(name = "branch")
+  private final String branch;
+
+  @Field(name = "builders")
```
Comment (holounic) [inline src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java:52]: Jackson (e.g ```@JsonProperty```) annotations are required for parsing jsons automatically. I would suggest to convert my POJO to yours before sending it to cloud storage.

**Review by holounic — APPROVED** [review]
LGTM!

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  @Unindexed
+  private final List<Builder> builders;
+
+  /**
+   * Converts a {@link #ParsedGitData ParsedGitData} object to a BuildInfo object.
+   * This is used for adding entries to the Google Cloud Datastore, from the Git commit
+   * information received, and leaving the {@code builders} field as empty, for later updates.
```
Comment (Somae) [inline src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java:59]:
```suggestion
   * information received, and leaving the {@code builders} field empty, for later updates.
```

**Review by Somae — APPROVED** [review]

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+import com.google.cloud.Timestamp;
+import java.util.ArrayList;
+import java.util.Date;
```
Comment (krasimir-google) [inline src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java:19]: nit: I believe the `java.util.Date` import is not needed anymore.

Comment (MagoDelBlocco) [reply src/main/java/com/google/graphgeckos/dashboard/storage/BuildInfo.java:19]: you are right!

**Pull request merged**